src/engine/thread.js

```diff
@@ -80,7 +99,14 @@ Thread.prototype.pushStack = function (blockId) {
     // Push an empty stack frame, if we need one.
     // Might not, if we just popped the stack.
     if (this.stack.length > this.stackFrames.length) {
+        // Copy warp mode from any higher level.
+        var warpMode = false;
+        if (this.stackFrames[this.stackFrames.length - 1]) {
+            warpMode = this.stackFrames[this.stackFrames.length - 1].warpMode;
+        }
         this.stackFrames.push({
+            isLoop: false, // Whether this level of the stack is a loop.
+            warpMode: warpMode, // Whether this level is in warp mode.
```

Comment:
Potential future change: it seems like storing `warpMode` in the stack frame gets complicated, in that there's more than one place (here, `goToNextBlock`, more?) which need to remember to copy the flag to a callee's stack frame.

Is there a safer way to track the flag, less prone to human error? Maybe as a function a argument? I don't want mile-long argument lists either, though. Just something to think about...